families/sawtooth_validator_registry/validator_registry/src/validator_registry_tp_verifier.rs

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2018 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+extern crate openssl;
+
+use self::openssl::{
+    hash::MessageDigest,
+    pkey::{PKey, Public},
+    sha::sha256,
+    sign::Verifier,
+};
+use crypto::{digest::Digest, sha2::Sha256};
+use protos::validator_registry::{SignUpInfo, SignUpInfoProof, ValidatorRegistryPayload};
+use sawtooth_sdk::{
+    messages::setting::Setting,
+    processor::handler::{ApplyError, TransactionContext},
+};
+
+use validator_registry_tp_handler::{get_state, parse_from};
+
+const _CONFIG_ADDRESS_PART_SIZE: usize = 16;
+const _CONFIG_NAMESPACE: &str = "000000";
+const _CONFIG_MAX_KEY_PARTS: usize = 4;
+
+pub fn verify_signup_info(
+    context: &mut TransactionContext,
+    originator_public_key_hash: &str,
+    val_reg_payload: &ValidatorRegistryPayload,
+) -> Result<(), ApplyError> {
+    let signup_info: SignUpInfo = val_reg_payload.get_signup_info().clone();
+    let proof_data: SignUpInfoProof = signup_info.get_proof_data().clone();
+
+    // Verify the attestation verification report signature
+    let verification_report = proof_data.get_verification_report();
+    let signature = &proof_data.get_signature();
+
+    // Try to get the report key from the configuration setting.  If it
+    // is not there or we cannot parse it, fail verification.
+    let report_public_key_pem =
+        get_config_setting(context, &"sawtooth.poet.report_public_key_pem".to_string())
+            .expect("Error reading config setting: PoET public key");
```

Comment:
Is using expect the right thing to do here and below? This will cause the tp to panic.

Reply:
Every expect in this function should be converted into the appropriate `InvalidTransaction` or `InternalError` error.  No place should use expect which, as @agunde406 said, will cause the TP to exit with a panic.